macros/tests.xml

```diff
@@ -0,0 +1,93 @@
+<macros>
```

Comment:
Doesn't this make it very unreadable for the actual tool?
And also here those macro files are shipped to all tools that import them, even if they do not use most of the stuff ... and it gets parsed and loaded into Galaxy I think.

Reply:
I'm torn, I like how compact it is, but I'm worried those files are growing and growing and single tools don't use much of it.

Reply:
Thanks for the feedback!

On the other hand, if the macros were split into more fine-grained files, this would result in more I/O operations for loading the tools. Do you think that this is tolerable and parsing a few superfluous macros is worse?

My concern is due to: I remember from my C++ days, that, for compilers, I/O operations are the reason while compile times are long. By concatenating all files into a single file (aka unitybuild), compile times could be reduced by >90% (the improvement was somewhat less when using SSD instead of HDD but still huge).

Reply:
I understand, but I guess this is not whats happening here. Every tool will ship the macro, so the number of files is the same in both cases. I one scenario its just a lot small files, and in your case its potentially a lot of big files with a lot of information that is not used for individual tools.

Its an academic discussion, please keep it like it is, I just wanted to point out this disadvantage.